src/embed_tests/TestDomainReload.cs

```diff
@@ -115,6 +116,71 @@ public static void CrossDomainObject()
             }
         }
 
+
+        #region Tempary tests
+        // https://github.com/pythonnet/pythonnet/pull/1074#issuecomment-596139665
+        [Test]
+        public void CrossReleaseBuiltinType()
+        {
+            try
+            {
+                var numRef = CreateNumReference();
+                GC.Collect();
+                GC.WaitForPendingFinalizers(); // <- this will put former `num` into Finalizer queue
+                Finalizer.Instance.Collect(forceDispose: true);
+                // ^- this will call PyObject.Dispose, which will call XDecref on `num.Handle`,
+                // but Python interpreter from "run" 1 is long gone, so it will corrupt memory instead.
+                Assert.False(numRef.IsAlive);
+            }
+            finally
+            {
+                PythonEngine.Shutdown();
+            }
+        }
+
+        [Test]
+        public void CrossReleaseCustomType()
+        {
+            try
+            {
+                var objRef = CreateConcreateObject();
+                GC.Collect();
+                GC.WaitForPendingFinalizers();
+                Finalizer.Instance.Collect(forceDispose: true);
+                Assert.False(objRef.IsAlive);
+            }
+            finally
+            {
+                PythonEngine.Shutdown();
+            }
+        }
+
+        private static WeakReference CreateNumReference()
+        {
+            PythonEngine.Initialize();
+            var num = 3216757418.ToPython();
+            Assert.AreEqual(num.Refcount, 1);
+            WeakReference numRef = new WeakReference(num, false);
+            PythonEngine.Shutdown(); // <- "run" 1 ends
+            PythonEngine.Initialize(); // <- "run" 2 starts
+            num = null;
+            return numRef;
+        }
+
+        private static WeakReference CreateConcreateObject()
+        {
+            PythonEngine.Initialize();
+            var obj = new Domain.MyClass().ToPython();
+            Assert.AreEqual(obj.Refcount, 1);
+            WeakReference numRef = new WeakReference(obj, false);
+            PythonEngine.Shutdown();
+            PythonEngine.Initialize();
+            obj = null;
+            return numRef;
+        }
+
+        #endregion Tempary tests
```

Comment:
Tests base from https://github.com/pythonnet/pythonnet/pull/1074#issuecomment-596139665

Reply:
Maybe move them to `pyinitialize.cs`?